src/remote/spectate_remote_server.ts

```diff
@@ -0,0 +1,182 @@
+      });
+
+      this.spectateRemoteServer = new WebSocketServer({ httpServer: this.httpServer });
+      this.spectateRemoteServer.on("request", async (request) => {
+        if (!this.connection && request.requestedProtocols.includes(SPECTATE_PROTOCOL)) {
+          const newConnection = request.accept(SPECTATE_PROTOCOL, request.origin);
+          const throttledRefresh = throttle(async () => {
+            try {
+              await this.spectateController!.refreshBroadcastList();
+            } catch (e) {
+              const err = typeof e === "string" ? e : e instanceof Error ? e.message : "unknown";
+              newConnection.sendUTF(JSON.stringify({ op: "list-broadcasts-response", err }));
+            }
+          }, 2000);
+          newConnection.on("message", async (data) => {
+            if (data.type === "binary") {
+              return;
+            }
+
+            const json = JSON.parse(data.utf8Data);
+            if (json.op === "list-broadcasts-request") {
+              await throttledRefresh();
+            } else if (json.op === "spectate-broadcast-request") {
+              const broadcastId = json.broadcastId;
+              if (!broadcastId || typeof broadcastId !== "string") {
+                newConnection.sendUTF(JSON.stringify({ op: "spectate-broadcast-response", err: "no broadcastId" }));
+                return;
+              }
+
+              const dolphinOptions: SpectateDolphinOptions = {};
+              const dolphinId = json.dolphinId;
+              if (dolphinId && typeof dolphinId === "string") {
+                dolphinOptions.dolphinId = dolphinId;
+              } else {
+                dolphinOptions.idPostfix = `remote${this.prefixOrdinal}`;
+                this.prefixOrdinal += 1;
+              }
+              try {
+                const path = this.settingsManager.get().settings.spectateSlpPath;
+                const dolphinId = await this.spectateController!.startSpectate(broadcastId, path, dolphinOptions);
+                newConnection.sendUTF(JSON.stringify({ op: "spectate-broadcast-response", dolphinId, path }));
+              } catch (e) {
+                const err = typeof e === "string" ? e : e instanceof Error ? e.message : "unknown";
+                newConnection.sendUTF(JSON.stringify({ op: "spectate-broadcast-response", err }));
+              }
+            }
+          });
+          newConnection.on("close", () => {
+            this.connection = null;
+            ipc_spectateRemoteStateEvent.main!.trigger({ connected: false, started: true }).catch(log.error);
+          });
+          this.connection = newConnection;
+          ipc_spectateRemoteStateEvent.main!.trigger({ connected: true, started: true }).catch(log.error);
+          this.connection.sendUTF(
+            JSON.stringify({
+              op: "spectating-broadcasts-event",
+              spectatingBroadcasts: await this.spectateController!.getOpenBroadcasts(),
+            }),
+          );
```

Comment:
Can some of these event handlers be moved out to their own private functions? This is a start method so ideally it should be fairly simple. i.e. start should "do start things", and each handler ideally should just be like:
```ts
newConnection.on("message", async (data) => this.handleOnMessage(data))
```
or something to that extent